Lib/unittest/mock.py

```diff
@@ -2403,8 +2408,12 @@ def __eq__(self, other):
         if self_name and other_name != self_name:
             return False
 
-        # this order is important for ANY to work!
-        return (other_args, other_kwargs) == (self_args, self_kwargs)
+        self_params = self_args, self_kwargs
+        other_params = other_args, other_kwargs
+        return (
+            self_params == other_params
```

Comment:
I agree with Paul's comment that this changes semantics. I am not sure if there is some third party code that depends on the order of equality for their own custom objects though as illustrated handles using `ANY` irrespective of order. I will wait for others opinion on this.

Reply:
Yeah, I agree this is the riskier of the two changes, and like I said, I'm happy to cherry-pick out the commits having to do with this if we just want to merge the other and hold off on this or think about it more. I'm having a hard time picturing someone using order of equality on calls for something purposeful, and an easy time imagining someone being confused that the two tests I added about this don't pass. But if someone was depending on this, it'd be really interesting to find out why. Maybe Paul will give me a hint on his more tightly scoped special casing `ANY` solution?

Reply:
I think I’ve got the special casing ANY solution, I’ll try to post it soon.